homeassistant/components/seventeentrack/__init__.py

```diff
@@ -1 +1,191 @@
+    """Unload SeventeenTrack Entry from config_entry."""
+
+    unload_ok = await hass.config_entries.async_unload_platforms(
+        config_entry, PLATFORMS
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(config_entry.entry_id)
+        if not hass.data[DOMAIN]:
+            hass.services.async_remove(DOMAIN, SERVICE_ADD_PACKAGE)
+            del hass.data[DOMAIN]
+
+    return unload_ok
+
+
+class SeventeenTrackDataCoordinator(DataUpdateCoordinator):
+    """Get the latest data from SeventeenTrack."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the data object."""
+        self.hass = hass
+        self.config_entry: ConfigEntry = config_entry
+        self.client: SeventeenTrackClient = None
+        self.account_id: str = ""
+        super().__init__(
+            self.hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_method=self.async_update,
+            update_interval=timedelta(
+                minutes=self.config_entry.options.get(
+                    CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                )
+            ),
+        )
+
+    @property
+    def show_archived(self) -> bool:
+        """Include archived packages when fetching data."""
+        return self.config_entry.options.get(CONF_SHOW_ARCHIVED, DEFAULT_SHOW_ARCHIVED)
+
+    async def async_update(self) -> dict[str, dict]:
+        """Update SeventeenTrack data."""
+        try:
+            packages = await self.client.profile.packages(
+                show_archived=False, tz=str(self.hass.config.time_zone)
+            )
+            summary = await self.client.profile.summary(show_archived=False)
+
+            if self.show_archived:
+                archived_packages = await self.client.profile.packages(
+                    show_archived=True, tz=str(self.hass.config.time_zone)
+                )
+                packages += archived_packages
+
+                archived_summary = await self.client.profile.summary(show_archived=True)
+                for status, qty in archived_summary.items():
+                    summary[status] += qty
+
+        except SeventeenTrackError as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+        _LOGGER.debug("New package data received: %s", packages)
+        _LOGGER.debug("New summary data received: %s", summary)
+
+        return {"packages": packages, "summary": summary}
+
+    async def async_setup(self) -> bool:
+        """Set up SeventeenTrack."""
+        try:
+            self.client = await get_client(self.hass, self.config_entry.data)
+            self.account_id = self.client.profile.account_id
+
+        except AuthenticationError as err:
+            raise ConfigEntryAuthFailed from err
+        except SeventeenTrackError as err:
+            raise ConfigEntryNotReady(
+                f"There was an error while logging in: {err}"
+            ) from err
```

Comment:
Its better to get the client under async_setup_entry then pass it to the coordinator.